File: Fragment/Fragment_demo_dong/app/src/main/java/com/example/fragment_demo_dong/Fragment_b.java
package com.example.fragment_demo_dong;

import android.graphics.Color;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

public class Fragment_b extends Fragment {


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_b, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        final EditText txtName=view.findViewById(R.id.name);
        Button bt=view.findViewById(R.id.bt);

        bt.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String age=txtName.getText().toString();
                txtName.setTextSize(30);
                txtName.setTextColor(Color.RED);
                txtName.setText("Age: "+age+" to old!");

            }
        });

    }
}

File: Fragment/Fragment_Api/app/src/main/java/com/example/fragment_api/Fragment_b.kt
package com.example.fragment_api

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.core.os.bundleOf
import androidx.fragment.app.setFragmentResult
import androidx.navigation.fragment.findNavController
import kotlinx.android.synthetic.main.fragment_a.*
import kotlinx.android.synthetic.main.fragment_b.*
import kotlinx.android.synthetic.main.fragment_a.edt_address as edt_address1


class Fragment_b : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_b, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        btn_send. setOnClickListener {
            try {
                val number = edt_address.text.toString().toInt()
                setFragmentResult (Fragment_a.REQUEST_KEY, bundleOf(ParentFragment.KEY_NUMBER to number))
            } catch (exception: NumberFormatException) {
                Toast.makeText(
                    requireContext(),
                    R.string.invalid_number_error,
                    Toast.LENGTH_SHORT
                ).show()
            }
        }
    }

//    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
//        super.onViewCreated(view, savedInstanceState)
//
//        val controller = findNavController()
//
//        btn_send.setOnClickListener {
//            val result = "result"
//            setFragmentResult("requestKey", bundleOf("bundleKey" to result))
//            Log.d("str", "onViewCreated: ")
//            controller.navigate(R.id.fragment_a)
//        }
//    }

//    class ParentFragment : Fragment(R.layout.fragment_parent) {
//        override fun onCreate(savedInstanceState: Bundle?) {
//            super.onCreate(savedInstanceState) childFragmentManager . setFragmentResultListener (REQUEST_KEY, this, FragmentResultListener { requestKey, result ->
//                val number =
//                    result.getInt(KEY_NUMBER) textParentNumber . text = getString (R.string.result_listener_number_message, number)
//            }) childFragmentManager.beginTransaction() .add(R.id.childFragment, ChildFragment::class.java, null) .commit()
//        }
//
//        companion object {
//            const val REQUEST_KEY = "parent-request-key" const
//                    val KEY_NUMBER = "key-number"
//        }
//    }

    class ChildFragment : Fragment(R.layout.fragment_child) {
        override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
            super.onViewCreated(
                view,
                savedInstanceState
            ) buttonChildDone . setOnClickListener {
                try {
                    val number = editTextChildNumber.text.toString()
                        .toInt() setFragmentResult (ParentFragment.REQUEST_KEY, bundleOf(ParentFragment.KEY_NUMBER to number))
                } catch (exception: NumberFormatException) {
                    Toast.makeText(
                        requireContext(),
                        R.string.invalid_number_error,
                        Toast.LENGTH_SHORT
                    ).show()
                }
            }
        }
    }

}